HW 3

MrFiszbi1 opened the pull request:
Hi, HW 3 is done. tell me what you think. I do want some feedback on certain parts. For ex-1 I saw hoping for some suggestions on a better way of implementing the hover method. my current way is kinda broken. For ex-2 I was hoping someone has a better sorting method than what I did. For the sandbox part on houses, I got my chart to work, but it does throw an error. any feedback on how I can fix it would be appreciated. Thanks all.

MrFiszbi committed:
HW 3

HW 3 Done. Code SandBox uploaded as zip and then unzipped to this branch

vpetrova123 commented on hw3/01-characters.js:
```diff
@@ -1,4 +1,71 @@
+  let name = document.createElement('figcaption');
+  let title = document.createElement('figcaption');
+
+  element.style.display = 'flex';
+  element.style.flexDirection = 'column';
+  element.style.alignItems = 'center';
+  element.style.width = '25%';
+  element.style.margin = '0 auto';
```
I think it would be a good idea to add a little bit of padding to the figure, especially on top.

vpetrova123 commented on hw3/01-characters.js:
```diff
@@ -1,4 +1,71 @@
+  Picture.style.height = '20vw';
+  name.textContent = character.fullName;
+  title.textContent = character.title;
+
+  element.addEventListener('mouseover', (event) => {
+    event.target.style.color = `rgb(255, 255, 255)`;
+    event.target.style.backgroundColor = '#800080';
+  });
+  element.addEventListener('mouseout', (event) => {
+    event.target.style.color = `initial`;
+    event.target.style.backgroundColor = 'initial';
+  });
+
```
To implement the hover in an easier way use the CSS hover effect.

```
figure:hover {
    background-color: #800080;
    color: rgb(255, 255, 255);
}
```

You should do all styling inside a CSS file instead of in the JS file, it's better practice.

vpetrova123 commented on hw3/02-charts.js:
I did something very similar for this one with the hard coded array of houses. Somebody suggested doing a dictionary instead of an array of houses and an array of houseCounter. That might help come up with a better sorting method.

vpetrova123 commented on hw3/ex03-05/src/Houses.jsx:
I'm not sure if I got the same error as you, but I got "chartData is not defined". I'm not sure if it's because you defined chartData inside the axios call, I know I've gotten weird errors when I do that. I moved the chart data inside function Houses and it worked for me, like this:
```
export default function Houses() {
  fetchData(url);

  const chartData = {
    labels: [
      'Targaryen',
      'Tarly',
      'Stark',
      'Baratheon',
      'Lannister',
      'Greyjoy',
      'Tyrell',
      'lesser House',
      'Free Folk',
      'No House',
    ],
    datasets: [
      {
        label: 'The Houses data',
        data: houseCounter,
        backgroundColor: backgroundColors,
        borderColor: borderColors,
        borderWidth: 1,
      },
    ],
  };

  return <Doughnut data={chartData} />;
}
```

vpetrova123 reviewed (commented):
Great job on the homework! Hopefully my suggestions were helpful

stevensonmat2 commented on hw3/ex03-05/src/Search.jsx:
```diff
@@ -0,0 +1,74 @@
+        />
+        {matchedCharacter && (
+          <figure>
+            <img
```
might be good to tell the user if no results are found.

stevensonmat2 commented on hw3/ex03-05/src/Houses.jsx:
don't know what I cant leave a normal comment on this file, but I would recommend an approach that programatically creates the labels for this. My approach was was to consolidate the characters into the correct houses, and then uses those houses for the label

stevensonmat2 commented on hw3/ex03-05/src/Search.jsx:
```diff
@@ -0,0 +1,74 @@
+  const [characters, setCharacters] = React.useState([]);
+  const [matchedCharacter, setMatchedCharacter] = React.useState(null);
+
+  React.useEffect(() => {
```
I would recommend using props to pass in the api data to each compoenent. that way you only have to make the call once

stevensonmat2 reviewed (commented):
looks good, just a few ideas

MrFiszbi committed:
HW 3 updated

updated based off code review. new hover implementation in styles.css, Houses.jsx uses useState to wait for chart to load.
